src/Services/ProductRegistrationService.php

```diff
@@ -55,21 +56,27 @@ public function getProductStatus(): array
                 }
             }
         } else {
-            // No change in version, so do not show the dialog if has opted out
-            if ($telemetry_disabled === null || $optOut == null) {
+            // No change in version, so do not show the dialog if has opted out of both email and telemetry
+            //  (if do show the dialog, then return if show email and/or telemetry dialog)
+            if ($telemetry_disabled == null || $optOut == null) {
                 $row['allowRegisterDialog'] = 1;
-            }
-            if ($optOut == null) {
-                $row['allowEmail'] = 1;
-            }
-            if (empty($telemetry_disabled)) {
-                $row['allowTelemetry'] = 1;
+                if ($optOut == null) {
+                    $row['allowEmail'] = 1;
+                }
+                if ($telemetry_disabled == null) {
+                    $row['allowTelemetry'] = 1;
```

Comment:
@bradymiller In current actions, $row['allowTelemetry'] should be enabled if $telemetry_disabled == 0 otherwise, none of the event collection routines will run to report telemetry.
Recommend a new flag collectTelemetryEvents or something. It'd be used in three places currently portal JS, Top Nav from tabs_view_model.js and API dispatcher.

Reply:
Just a reminder to remove the exposed unused 
```
case 'allUsageData':
            $result = $telemetryService->reportUsageData();
            echo json_encode($result);
            break;
```

In track_portal_events.php

Reply:
will remove that chunk of code in commit to PR tonight :)

Reply:
The three places that assess if telemetry is disabled now use your handydandy TelemetryService::isTelemetryEnabled() call :)

Reply:
when in doubt, write a method!:)